src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -265,7 +276,9 @@ public override bool IsInvalid
 
         protected override bool ReleaseHandle()
         {
-            Interop.OpenSsl.FreeSslContext(_sslContext);
+            Interop.OpenSsl.Disconnect(_sslContext);
+            _sslContext.DangerousRelease();
+            _sslContext.Dispose();
```

Comment:
I'm not sure about this Dispose.  It seems like you're still in the exact same situation as right now.

It seems like this Dispose should be called from `public override void Dispose(bool disposing)` only when `disposing` is true.  Not ReleaseHandle unconditionally.  (The `DangerousRelease` does belong in ReleaseHandle, though, since it should happen on both Dispose and Finalize).